src/gui/sdlmain.cpp

```diff
@@ -2392,9 +2385,13 @@ void GFX_EndUpdate(const uint16_t *changedLines)
 		case FrameMode::Cfr:
 			maybe_present_synced(frame_is_new);
 			break;
-		case FrameMode::Vfr: present_new_or_maybe_dupe(frame_is_new); break;
+		case FrameMode::Vfr:
+			if (frame_is_new) {
+				sdl.frame.present();
+			}
+			break;
 		case FrameMode::ThrottledVfr:
-			maybe_present_throttled_or_dupe(frame_is_new);
+			maybe_present_throttled(frame_is_new);
```

Comment:
Yes, so this commit is only what we need from this PR, I think. And we _absolutely_ want this for _any_ screen mode, not just text modes!